server/oauth2.go

```diff
@@ -575,7 +588,8 @@ func validateRedirectURI(client storage.Client, redirectURI string) bool {
 	if err != nil {
 		return false
 	}
-	if u.Scheme != "http" {
+	// public clients should use http or https (#1300)
+	if u.Scheme != "http" && u.Scheme != "https" {
```

Comment:
This was discussed in open issue #1300 – it was a quick/logical quick so included here, but not relevant to main issue of wildcard support.